Design/PDcalibration.md

```diff
@@ -0,0 +1,145 @@
+Motivation
+==========
+
+Overall the calibration is being used to calculate the time-of-flight
+to d-spacing conversion factors. Using "GSAS" style parameters this is
+```
+TOF = (DIFC * d) + (DIFA * d * d) + TZERO
+```
+The preference is to use `DIFC`, then add `TZERO` and `DIFA` (in
+order) if necessary.
+
+Algorithms like
+[GetDetectorOffsets](https://github.com/mantidproject/mantid/blob/v3.5.1/Code/Mantid/Framework/Algorithms/src/GetDetectorOffsets.cpp)
+and
+[GetDetOffsetsMultiPeaks](https://github.com/mantidproject/mantid/blob/v3.5.1/Code/Mantid/Framework/Algorithms/src/GetDetOffsetsMultiPeaks.cpp)
+use the instrument geometry to convert the calibration data from
+time-of-flight (`TOF`) to d-spacing then use cross-correlation or
+single peak fitting to determine an `offset` which is nominally a
+percentage difference of the conversion using the nominal `DIFC` and
+what is actually needed.
+
+There are a few reasons why this is being done:
+
+1. NOMAD data does not align/focus well enough using a simple
+   `DIFC`/`offset` paradigm.
+
+1. Current calibration happens "from scratch" using the instrument
+   geometry as a starting point. This means that every calibration is
+   (essentially) a global search.
+
+1. Calibration requires a "long soak" type measurement where all of
+   the pixels have enough statistics for either peak fitting or
+   cross-correlation. This requires a fair amount of beamtime for
+   things that could be handled with a smaller (multiplicative)
+   correction.
+
+Requirements
+============
+
+1. Two user facing algorithms for calibrating the instrument.
+
+   * Algorithm that performs a pixel-by-pixel calibration by using a
+     "many peaks" approach. GetDetOffsetsMultiPeaks
+     ([docs](http://docs.mantidproject.org/v3.5.1/algorithms/GetDetOffsetsMultiPeaks-v1.html)
+     and
+     [github](https://github.com/mantidproject/mantid/blob/v3.5.1/Code/Mantid/Framework/Algorithms/src/GetDetOffsetsMultiPeaks.cpp))
+     is representative.
+
+   * Algorithm that performs bank-by-bank calibration to generate a
+     multiplicative factor for each summed
+     spectra. [EnggCalibrate](https://github.com/mantidproject/mantid/blob/v3.5.1/Code/Mantid/Framework/PythonInterface/plugins/algorithms/EnggCalibrate.py)
+     is representative of this approach.
+
```

Comment:
How often do they do the long run for pixel-by-pixel calibration on NOMAD?
With ENGIN-X the very long calibration run is done every ~[3, 5] years only. Then the bank-by-bank calibration runs are typically done once or twice per cycle or few months.
For ISIS powder diffraction instruments the current practice is apparently to re-use .cal files for a few cycles or around one year. I'd suspect this must be similar as for NOMAD. But we don't have this properly under control at present.